x-pack/plugins/fleet/public/applications/fleet/sections/agents/components/agent_upgrade_modal/index.test.tsx

```diff
@@ -180,10 +189,115 @@ describe('AgentUpgradeAgentModal', () => {
       );
       expect(optionList.textContent).toEqual(['8.10.4', '8.10.2+build123456789'].join(''));
     });
+
+    it('should disable submit button and display a warning for a single agent when version is greater than maxFleetServerVersion', async () => {
+      mockSendGetAgentsAvailableVersions.mockClear();
+      mockSendGetAgentsAvailableVersions.mockResolvedValue({
+        data: {
+          items: ['8.10.4', '8.10.2', '8.9.0', '8.8.0'],
+        },
+      });
+      mockSendAllFleetServerAgents.mockResolvedValue({
+        allFleetServerAgents: [
+          { id: 'fleet-server', local_metadata: { elastic: { agent: { version: '8.9.0' } } } },
+        ] as any,
+      });
+
+      const { utils } = renderAgentUpgradeAgentModal({
+        agents: [
+          {
+            id: 'agent1',
+            local_metadata: {
+              elastic: {
+                agent: { version: '8.8.0', upgradeable: true },
+              },
+              host: { hostname: 'host00001' },
+            },
+          },
+        ] as any,
+        agentCount: 1,
+      });
+
+      await waitFor(() => {
+        const container = utils.getByTestId('agentUpgradeModal.VersionCombobox');
+        const input = within(container).getByRole<HTMLInputElement>('combobox');
+        expect(input?.value).toEqual('8.10.2');
+        expect(
+          utils.queryAllByText(
+            /This action will upgrade the agent running on 'host00001' to version 8.10.2. This action can not be undone. Are you sure you wish to continue?/
+          )
+        );
+        expect(utils.queryByText(/Error upgrading the selected agent/)).toBeInTheDocument();
```

Comment:
I think it would be better to use a warning instead of error, since this is not a result of an action, but preventing it: https://eui.elastic.co/#/display/callout#warning
Could have a title something like `Action disallowed` or `Warning`

Reply:
+1 - let's go with a warning callout instead of an error here.

Reply:
I repurposed the callout added in https://github.com/elastic/kibana/pull/173253 as those conditions are exclusive. Also updated the copy so the callout will just display "warning" as title and whatever message is relevant for the case.
I still need to do some cleanup on the component as there is some code duplication and the modal is now doing a lot of things.